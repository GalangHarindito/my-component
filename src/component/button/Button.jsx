import React from "react";
import PropTypes from "prop-types";
import "./button.scss";
import { Loader } from "../spinner/loader";

export const Button = ({ mode, backgroundColor, size, label, loading, disabled, icon, onClick, ...props }) => {

  return (
    <button
      type="button"
      className={["storybook-button", `storybook-button--${size}`, `${mode && `storybook-button--${mode}`}`, `${disabled && `disabled`}`].join(
        " "
      )}
      onClick={onClick}
      style={backgroundColor && { backgroundColor }}
      disabled = {disabled}
      {...props}
    >
      <div className={loading || icon ? 'button-aside' : ''}>
      {loading ? <Loader size='small' />  : '' }
      {icon}
      <span>{loading? 'Loading': label }</span>
      </div>
    </button>
  );
};

Button.propTypes = {
  mode: PropTypes.oneOf(["primary","secondary", "info", "warning", "danger"]),
  backgroundColor: PropTypes.string,
  size: PropTypes.oneOf(["small", "medium", "large"]),
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  mode: "primary",
  backgroundColor: null,
  size: "large",
  onClick: undefined,
  loading: false,
  disabled: false,
  label:"Button"
};
