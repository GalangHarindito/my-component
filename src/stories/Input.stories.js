import { Input } from "../component/input/Input";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Input',
  component: Input,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    onChange: {
      description: 'Handle behaviour when input event',
      action: 'type',
    },
    placeholder: {
      description: 'Name for placeholder',
      action: 'string',
    },
    label: {
      description: 'Name for input label',
      action: 'string',
    },
    size: {
      description: 'Size for input',
      action: 'string',
    },
    error: {
      description: 'Error message if any',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
    label: 'Label Text',
    placeholder: 'Input Name',
    error: ''
  },
};

export const Medium = {
    args: {
      size: 'medium',
      label: 'Label Text',
      placeholder: 'Input Name',
      error: ''
    },
  };