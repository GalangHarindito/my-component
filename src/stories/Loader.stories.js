import { Loader } from "../component/spinner/loader";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Loader',
  component: Loader,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    size: {
      description: 'Size for loader',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
  },
};

export const Medium = {
    args: {
      size: 'medium',
    },
  };

  export const Small = {
    args: {
      size: 'small',
    },
  };