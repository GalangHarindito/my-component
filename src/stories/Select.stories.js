import { SelectInput } from "../component/select/select";

export default {
    title: 'Component/Select',
    component: SelectInput,
    parameters: {
      // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
      layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
    tags: ['autodocs'],
    // More on argTypes: https://storybook.js.org/docs/api/argtypes
  
    argTypes:{
        onChange: {
            description: 'Handle behaviour when input event',
            action: 'type',
          },
          label: {
            description: 'Name for input label',
            action: 'string',
          },
          size: {
            description: 'Size for input',
            action: 'string',
          },
          error: {
            description: 'Error message if any',
            action: 'string',
          },
          required: {
            description: 'Required field',
            action: 'boolean',
          },
    }
  };
  
  // More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
  
  export const Select = {
    args: {
      size: 'large',
      label: 'Label Text',
      defaultValue: ' Input Value',
      error: '', 
      required: false,
      options: [
        { icon: "ðŸ¤£", value: "test1", name: "test1" },
        { icon: "ðŸ˜†", value: "test2", name: "test2" },
        { icon: "ðŸ¥°", value: "test3", name: "test3" },
      ]
    },
  };