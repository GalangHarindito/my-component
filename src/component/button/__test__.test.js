import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../../stories/Button.stories';
import { Button } from './Button';
import userEvent from '@testing-library/user-event'

const { Primary, Secondary } = composeStories(stories);

test('Button layut should not change', () => {
    const { container } = render(<Primary />)
    expect(container).toMatchSnapshot()
})

test('Button layout should not change', () => {
    const { container } = render(<Secondary />)
    expect(container).toMatchSnapshot()
})

test('renders primary button with default args', () => {
  render(<Primary />);
  const buttonElement = screen.getByText(
    /Button/i
  );
  expect(buttonElement).not.toBeNull();
});

test('renders primary button with overriden props', () => {
  render(<Primary>Button</Primary>); // you can override props and they will get merged with values from the Story's args
  const buttonElement = screen.getByText(/Button/i);
  expect(buttonElement).not.toBeNull();
});

test('renders button on click event', () => {
  const onChangeSpy = jest.fn()

  render(<Button onChange={onChangeSpy} label='label' />)
  const label = screen.getByText(/label/i)
  const button = screen.getByRole('button')
  userEvent.click(button)

  expect(label).not.toBeNull()
  expect(onChangeSpy).not.toHaveBeenCalled()
})

test('renders secondary button with default args', () => {
    render(<Secondary />);
    const buttonElement = screen.getByText(
      /Button/i
    );
    expect(buttonElement).not.toBeNull();
  });
  
  test('renders primary button with overriden props', () => {
    render(<Secondary>Button</Secondary>); // you can override props and they will get merged with values from the Story's args
    const buttonElement = screen.getByText(/Button/i);
    expect(buttonElement).not.toBeNull();
  });