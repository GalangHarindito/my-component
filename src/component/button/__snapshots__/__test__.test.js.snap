// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button layout should not change 1`] = `
<div>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`Button layut should not change 1`] = `
<div>
  <button
    class="storybook-button storybook-button--medium storybook-button--primary"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`Button layut should not change 2`] = `
<div>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</div>
`;
